src/model/OWL.pl

```diff
@@ -405,7 +431,7 @@
   has_equivalent_class(Cls,EQ).
 
 %%
-has_equivalent_class1([Entity|_],Entity,_) :- !.
+has_equivalent_class1([],Visited,Visited).
```

Comment:
I do not get why second/third argument must be the same in that case.
Shouldn't the second argument be also an empty list then?
As it carries the list output, I think it should be empty list in that clause.

Reply:
I use visited to ¨collect¨ all equivalent classes. This behaviour is used at the last line of the next case (ln. 443).

Reply:
hmm so visited has only equivalent classes in it?

Reply:
could we then remove the second argument entirely?

Reply:
It's not obvious to me, how it would be done  without the argument. The third argument is used to collect the visited equivalent classes and then we bin (in the marked case here) this "collection" to the second variable. But Visited has to be an empty list in the highest call.